packages/core/test/types.test.ts

```diff
@@ -3193,4 +3193,45 @@ describe('tags', () => {
     // @ts-expect-error
     actor.getSnapshot().hasTag('other');
   });
+
+  it('supports partial event descriptor types', () => {
```

Comment:
I'd prefer to have separate tests for each case. It helps long-term because each case receives a unique test title so it becomes obvious what exact requirement is not met when something breaks.

At the same time, currently - all of this testing happens in a single function call. I've seen in the past situations in which one error could accidentally impact another one, some error location could get moved, or perhaps only a subset of errors was actually reported. Either way, it's safer~ to just have such things isolated in granular tests.